canopeum_frontend/src/pages/Login.tsx

```diff
@@ -1,150 +1,116 @@
 
-const Login = () => {
-  const { authenticate } = useContext(AuthenticationContext)
-  const { t: translate } = useTranslation()
-  const { getApiClient } = useApiClient()
-
-  const [email, setEmail] = useState('')
-  const [password, setPassword] = useState('')
-  const [rememberMe, setRememberMe] = useState(false)
-
-  const [emailError, setEmailError] = useState<InputValidationError | undefined>()
-  const [passwordError, setPasswordError] = useState<InputValidationError | undefined>()
-
-  const [loginError, setLoginError] = useState<string | undefined>()
-
-  const validateEmail = () => {
-    if (!email) {
-      setEmailError('required')
-
-      return false
-    }
-
-    setEmailError(undefined)
-
-    return true
-  }
-
-  const validatePassword = () => {
-    if (!password) {
-      setPasswordError('required')
-
-      return false
-    }
-
-    setPasswordError(undefined)
-
-    return true
-  }
-
-  const validateForm = () => {
-    // Do not return directly the method calls;
-    // we need each of them to be called before returning the result
-    const emailValid = validateEmail()
-    const passwordValid = validatePassword()
-
-    return emailValid
-      && passwordValid
-  }
+type LoginFormInputs = {
+  email: string,
+  password: string,
+  rememberMe: boolean,
+}
 
-  const onLoginClick = async () => {
-    const isFormValid = validateForm()
-    if (!isFormValid) return
+const Login = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, touchedFields },
+  } = useForm<LoginFormInputs>({ mode: 'onTouched' })
 
+  const onSubmit: SubmitHandler<LoginFormInputs> = async formData => {
     try {
       const response = await getApiClient().authenticationClient.login(
         new LoginUser({
-          email: email.trim(),
-          password,
+          email: formData.email.trim(),
+          password: formData.password,
         }),
       )
 
-      storeToken(response.token, rememberMe)
+      storeToken(response.token, formData.rememberMe)
       authenticate(response.user)
     } catch {
       setLoginError(translate('auth.log-in-error'))
     }
   }
 
+  const { authenticate } = useContext(AuthenticationContext)
+  const { t: translate } = useTranslation()
+  const { getApiClient } = useApiClient()
+
+  const [loginError, setLoginError] = useState<string | undefined>()
+
   return (
     <AuthPageLayout>
       <div style={{ flexGrow: '0.4', display: 'flex', alignItems: 'center' }}>
         <h1 style={{ textAlign: 'center' }}>{translate('auth.log-in-header-text')}</h1>
       </div>
 
-      <div className='col-10 col-sm-6 col-md-8 col-xl-6 d-flex flex-column gap-4'>
+      <form
+        className='col-10 col-sm-8 col-xl-6 d-flex flex-column gap-4'
+        onSubmit={handleSubmit(onSubmit)}
+      >
         <div className='w-100'>
-          <label htmlFor='email-input'>{translate('auth.email-label')}</label>
+          <label htmlFor='email'>{translate('auth.email-label')}</label>
```

Comment:
The label's `htmlFor` needs to point to the associated element's id.

This isn't just semantics, it has a functional difference where interacting with the label (like a click) will interract / focus the input element itself.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/for